src/common/utils/pagination.ts

```diff
@@ -0,0 +1,55 @@
+import { List, PaginationOptions } from '../interfaces';
+
+export class AutoPaginatable<T> {
+  readonly object: 'list' = 'list';
+  readonly options: PaginationOptions;
+
+  constructor(
+    private list: List<T>,
+    private apiCall: (params: PaginationOptions) => Promise<List<T>>,
+    options?: PaginationOptions,
+  ) {
+    this.options = {
```

Comment:
Is this any different from `this.options = options`?

Reply:
Seems worth simplifying. We can always change it back later.

Reply:
I get an undefined error if I don't use the spread operator, not sure if there's another way around that I should be taking.